src/common/charts/LegendChart.jsx

```diff
@@ -70,7 +72,7 @@ module.exports = React.createClass({
       <div style={{'width': props.width, 'height': props.height}} >
         {this._renderTitle()}
         {this._renderLegend()}
-        <svg viewBox={props.viewBox} width={props.width - props.sideOffset} height={props.height}>{props.children}</svg>
+        <svg viewBox={this.getViewBox()} width={this.getOuterDimensions().width - props.sideOffset} height={props.height}>{props.children}</svg>
```

Comment:
> I'm not sure how you could do a table or float layout with all the SVG we use for axes. SVG unfortunately doesn't support any kind of flow layout.

We'd only need CSS layout to get the legend (a `div`) to flow alongside of fluid-width `svg` elements.

The overall approach taken in this PR is solid and has achieved fluid-width charts for us (@samdalton and I), the implementation just currently has a bug in `LegendChart`.